### ISSUE TITLE
:escape_html option overridden in Rails 'production' environment

### ISSUE THREAD

**Issue by wjordan** [haml/haml#980]
This issue was originally reported by @Justin-Maxwell in https://github.com/haml/haml/issues/940#issuecomment-325833327.

It seems that starting with version 5 of `haml`, some lazy-loading behavior changed so that the `escape_haml` option set in a Rails initializer will be re-set to `true` in non-`development` environments (e.g., `production`).

As mentioned in the linked issue-comment, this is a relatively dangerous bug because it can produce working code in development that breaks only when executed in production.

Here is a minimal demonstration Bash script that reproduces the issue consistently:

`haml.sh`:
```bash
#!/bin/bash

RAILS_PATH=${RAILS_PATH:-/tmp/haml}
RAILS_VERSION=${RAILS_VERSION:-5.0.6}
HAML_VERSION=${HAML_VERSION:-'~> 4'}

rails _${RAILS_VERSION}_ new ${RAILS_PATH} -q > /dev/null

pushd ${RAILS_PATH} > /dev/null
echo 'Haml::Template.options[:escape_html] = false' > config/initializers/haml.rb
echo "gem 'haml', '${HAML_VERSION}'" >> Gemfile
bundle install --quiet
DISABLE_SPRING=1 RAILS_ENV=development bin/rails r 'puts "development: #{Haml::Template.options[:escape_html]}"'
DISABLE_SPRING=1 RAILS_ENV=production bin/rails r 'puts "production: #{Haml::Template.options[:escape_html]}"'
popd > /dev/null
```

```
$ echo 'haml4:'; HAML_VERSION='~> 4' ./haml.sh; echo 'haml5:'; HAML_VERSION='~> 5' ./haml.sh
haml4:
development: false
production: false
haml5:
development: false
production: true
```

**Comment by curious725** [comment]
@wjordan Hello,
as I can see you already fixed this issue in #984,
please close this issue.
I spent a lot of time digging into it.

**Comment by k0kubun** [comment]
sure.
